catalog/spatial/registry/registry-federation-admin-service-impl/src/main/java/org/codice/ddf/registry/federationadmin/service/impl/RefreshRegistrySubscriptions.java

```diff
@@ -114,13 +114,18 @@ private void addOrUpdateMetacardsForRemoteRegistries(Set<String> sourceIds)
     private Map<String, Metacard> getRegistryMetacardsMap() throws FederationAdminException {
 
         Map<String, Metacard> registryMetacardsMap = new HashMap<>();
-        List<Metacard> registryMetacards = federationAdminService.getRegistryMetacards();
+        try {
+            List<Metacard> registryMetacards = javax.security.auth.Subject.doAs(subject,
```

Comment:
Who calls `refreshRegistrySubscriptions()`? When elevating privileges, it'd be best to not do so in a public method, but instead contain that to something that is running internally. For example, throwing in a class that polls to refresh subscriptions automatically and, internally, uses the system subject. Then other pieces of code can't call some public method to assume the new identity.